osquery/tables/events/linux/process_events.cpp

```diff
@@ -51,124 +84,262 @@ Status AuditProcessEventSubscriber::ProcessEvents(
     CopyFieldFromMap(row, syscall_event_record->fields, "euid", "0");
     CopyFieldFromMap(row, syscall_event_record->fields, "gid", "0");
     CopyFieldFromMap(row, syscall_event_record->fields, "egid", "0");
-    CopyFieldFromMap(row, cwd_event_record->fields, "cwd", "0");
 
-    std::uint64_t parent_process_id;
-    GetIntegerFieldFromMap(
-        parent_process_id, syscall_event_record->fields, "ppid");
-    row["parent"] = std::to_string(parent_process_id);
+    if (!GetSyscallName(row["syscall"], event_data.syscall_number)) {
+      row["syscall"] = std::to_string(event_data.syscall_number);
+      VLOG(1) << "Failed to locate the system call name";
+    }
 
     std::string field_value;
     GetStringFieldFromMap(field_value, syscall_event_record->fields, "exe", "");
     row["path"] = DecodeAuditPathValues(field_value);
 
-    auto qd = SQL::selectAllFrom("file", "path", EQUALS, row.at("path"));
-    if (qd.size() == 1) {
-      row["ctime"] = qd.front().at("ctime");
-      row["atime"] = qd.front().at("atime");
-      row["mtime"] = qd.front().at("mtime");
-      row["btime"] = "0";
+    // Acquire the process IDs
+    std::uint64_t parent_process_id{0U};
+    std::uint64_t process_id{0U};
+    auto s = GetProcessIDs(parent_process_id,
+                           process_id,
+                           event_data.syscall_number,
+                           *syscall_event_record);
+    if (!s.ok()) {
+      VLOG(1) << "Malformed AUDIT_SYSCALL event: " << status.getMessage();
+      continue;
     }
 
+    row["parent"] = std::to_string(parent_process_id);
+    row["pid"] = std::to_string(process_id);
+
+    // Unused fields
     row["overflows"] = "";
+    row["env"] = "";
     row["env_size"] = "0";
     row["env_count"] = "0";
-    row["env"] = "";
-    row["uptime"] = std::to_string(getUptime());
 
-    // build the command line from the AUDIT_EXECVE record
-    row["cmdline"] = "";
-
-    for (const auto& arg : execve_event_record->fields) {
-      if (arg.first == "argc") {
+    // Handle syscall-specific data
+    if (is_exec_syscall) {
+      auto status = ProcessExecveEventData(row, event);
+      if (!status) {
+        VLOG(1) << "Failed to parse the event: " << status.getMessage();
         continue;
       }
 
-      // Amalgamate all the "arg*" fields.
-      if (row.at("cmdline").size() > 0) {
-        row["cmdline"] += " ";
-      }
+    } else {
+      row["owner_uid"] = "0";
+      row["owner_gid"] = "0";
+    }
+
+    emitted_row_list.push_back(row);
+  }
+
+  return Status::success();
+}
+
+Status AuditProcessEventSubscriber::ProcessExecveEventData(
+    Row& row, const AuditEvent& event) noexcept {
+  // Get the initial working directory of the process
+  const AuditEventRecord* cwd_event_record = GetEventRecord(event, AUDIT_CWD);
+  if (cwd_event_record == nullptr) {
+    return Status::failure("Malformed AUDIT_CWD event");
+  }
+
+  CopyFieldFromMap(row, cwd_event_record->fields, "cwd", "");
```

Comment:
I see common calls with `""` and `"0"` as the third argument. Is it worth making this API more explicit with an overload + fall through?

Reply:
Those arguments are the default value to adopt in case the specified field is missing in the field map, and varies depending on the context. How would you change this?